### ISSUE TITLE
MacOs 14.3 Right click/Open File with mrv2 results in error

### ISSUE THREAD

**Issue by piotrnikov** [ggarra13/mrv2#184]
Hello!
First of all thanks so much for a great player!!
And second, I have an issue on Mac. When right clicking on a video file and choosing mrv2 as a program to open the file, it results in the following error:
The document “Blabla.mp4” could not be opened. mrv2 cannot open files in the “MPEG-4 movie” format.
but when I drag and drop the same file into mrv2 it opens just fine...
Any way to fix this?
Thanks again!!

Piotr

**Comment by ggarra13** [comment]
> When right clicking on a video file and choosing mrv2 as a program to open the file, it results in the following error:
> The document “Blabla.mp4” could not be opened. mrv2 cannot open files in the “MPEG-4 movie” format.
> but when I drag and drop the same file into mrv2 it opens just fine...
> Any way to fix this?

Sadly, there's nothing I can do.  It is a bug/misfeature of macOS.  The problem is that mrv2 is not a registered application with Apple (which costs around U$200 per year) and as such results in that error in addition to the trouble of installing it.
You might want to report it with Apple to see if they fix it.

**Comment by p2or** [comment]
Hi folks,

bummer! Just out of curiosity: Why did it work perfectly with _mrViewer_? Was this introduced as of MacOS 13? My current workaround is to create a *wrapper* with [Automator](https://support.apple.com/guide/automator/welcome/mac) using a shell script which seems to work for all file types:

![Screenshot 2024-02-25 at 10 10 21](https://github.com/ggarra13/mrv2/assets/512368/094857a4-6c80-4f81-a0ce-42f499026010)

Extract the archive, copy `mrv2.sh.app` e.g. into the 'Applications' folder and make this the default: [mrv2.sh.zip](https://github.com/ggarra13/mrv2/files/14395856/mrv2.sh.zip)



Cheers,
Christian

**Comment by p2or** [comment]
Hi @ggarra13,

> It was introduced I think in macOS 11 mrViewer was written before Apple implemented their certification program and as such it is able to run fine even today ...

interesting, thank you. [Automator](https://support.apple.com/guide/automator/welcome/mac) comes with the system - I don't think it's necessary to install anything.

Cheers,
Christian

**Comment by ggarra13** [comment]
@p2or @piotrnikov Version 1.4.5 re-release just now has a proper fix to support Open With.  There was no need to notarize the application contrary to the resources I had found back when I posted this.

**Comment by p2or** [comment]
Hi @ggarra13,

unfortunately, I cannot confirm. With the actual release I downloaded 14 hours ago, I am receiving the same error (it works with my shell script however):

<img width="935" height="705" alt="Image" src="https://github.com/user-attachments/assets/202d3c92-7f4b-425b-95c6-f3469124e69d" />

----

The **re-release** version from 8 hours ago crashes immediately so I can't really test: 

```
Process:               mrv2 [17649]
Path:                  /Applications/mrv2.app/Contents/MacOS/mrv2
Identifier:            com.mrv2.mrv2
Version:               1.4.5 (1.4.5)
Code Type:             ARM-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2025-11-09 08:23:39.9622 +0100
OS Version:            macOS 15.6.1 (24G90)
Report Version:        12
Anonymous UUID:        FA1F6862-B7C0-81AE-345F-E7B6CE90233E

Sleep/Wake UUID:       6757A012-536D-491C-8A7A-6EEAD80D7B75

Time Awake Since Boot: 100000 seconds
Time Since Wake:       1891 seconds

System Integrity Protection: enabled

Dyld Error Message:
  1

Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000

Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: @rpath/libfltk_images.1.5.dylib
Referenced from: <73C06AB1-7D5C-3D95-BC1A-FE9B6BEC32CC> /Applications/mrv2.app/Contents/MacOS/mrv2
Reason: tried: '/Users/runner/work/mrv2/mrv2/BUILD-Darwin-arm64/Release/install/lib/libfltk_images.1.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/runner/work/mrv2/mrv2/BUILD-Darwin-arm64/Release/install/lib/libfltk_images.1.5.dylib' (no such file), '/Users/runner/work/mrv2/mrv2/BUILD-Darwin-arm64/Release/install/lib/libfltk_images.1.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/runner/work/mrv2/mrv2/BUILD-Darwin-arm64/Release/install/lib/libfltk_images.1.5.dylib' (no such file)
(terminated at launch; ignore backtrace)
```
I have to work in the upcoming days, but if I can test anything in one or two weeks, just let me know.

Cheers,
Christian

**Comment by ggarra13** [comment]
Okay.  We made progress.  I've updated v1.4.5 again and tested as far as I could and it seems to work.

Please give it a try from Github.

**Comment by p2or** [comment]
Hi @ggarra13,

sure:

```sh
(base) p2or@mbp ~ % otool -l /Applications/mrv2.app/Contents/Resources/bin/mrv2 | grep -A 2 LC_RPATH
          cmd LC_RPATH
      cmdsize 104
         path @executable_path/../lib:/Users/runner/work/mrv2/mrv2/BUILD-Darwin-arm64/Release/install/lib (offset 12)
--
          cmd LC_RPATH
      cmdsize 32
         path /usr/local/lib (offset 12)
--
          cmd LC_RPATH
      cmdsize 80
         path /Users/runner/work/mrv2/mrv2/BUILD-Darwin-arm64/Release/install/lib (offset 12)
```

Does this help?

**Comment by ggarra13** [comment]
Yes.   That helps.  You don't have the latest version installed or something got mixed up. 

This is what you should have gotten (notice executable_path and /Users/runner/work are two different lines not one line separated with colon as in what you showed) 

```
$ otool -l /Applications/mrv2.app/Contents/Resources/bin/mrv2 | grep -A2 LC_RPATH

          cmd LC_RPATH
      cmdsize 40
         path @executable_path/../lib (offset 12)
--
          cmd LC_RPATH
      cmdsize 80
         path /Users/runner/work/mrv2/mrv2/BUILD-Darwin-amd64/Release/install/lib (offset 12)
--
          cmd LC_RPATH
      cmdsize 32
         path /usr/local/lib (offset 12)
```

**Comment by ggarra13** [comment]
> Good catch, thanks! Finally, I can confirm that it now works on right-click as expected. However, it flickers briefly after execution, but I guess this is due to the new license window, right? Will this window last forever from now on?
> 
> Cheers, Christian

Not sure what you.  It may be due to the licensing window.  There was also a minor problem in mrv2 (not vmrv2) where a minimized shell window would get opened right next to the mrv2 icon.  That will get fixed on v1.4.6 next week.

P.S.  If you want to get rid of the licensing window at start and every 3 mins, you really need to donate either u$50 (for one year) or u$150 (to own for 100 years ;).
